pkg/cmd/server/api/v1/types.go

```diff
@@ -702,93 +702,100 @@ type LDAPSyncConfig struct {
 	// If empty, the default system roots are used
-	CA string `json:"CA,omitempty" description:"an optional trusted CA to use when making requests to the server"`
+	CA string `json:"ca" description:"an optional trusted CA to use when making requests to the server"`
 
 	// LDAPGroupUIDToOpenShiftGroupNameMapping is an optional direct mapping of LDAP group UIDs to
 	// OpenShift Group names
-	LDAPGroupUIDToOpenShiftGroupNameMapping map[string]string
+	LDAPGroupUIDToOpenShiftGroupNameMapping map[string]string `json:"ldapGroupUIDToOpenShiftGroupNameMapping" description:"optional direct mapping of LDAP group UIDs to OpenShift Group names"`
 
-	// LDAPSchemaSpecificConfig holds the configuration for retrieving data from the LDAP server.
-	// This set of configuration varies with LDAP server schema.
-	LDAPSchemaSpecificConfig `json:"inline,omitempty" description:"schema-specific LDAP client configuration"`
-}
-
-// LDAPSchemaSpecificConfig holds the schema-specific configuration for data retrieval from the LDAP
-// server. Only one of the members can be specified.
-type LDAPSchemaSpecificConfig struct {
 	// RFC2307Config holds the configuration for extracting data from an LDAP server set up in a fashion
 	// similar to RFC2307: first-class group and user entries, with group membership determined by a
 	// multi-valued attribute on the group entry listing its' members
-	RFC2307Config *RFC2307Config `json:"RFC2307,omitempty" description:"schema-specific information for an RFC2307-like schema"`
+	RFC2307Config *RFC2307Config `json:"rfc2307" description:"schema-specific information for an RFC2307-like schema"`
 
 	// ActiveDirectoryConfig holds the configuration for extracting data from an LDAP server set up in a
 	// fashion similar to that used in Active Directory: first-class user entries, with group membership
 	// determined by a multi-valued attribute on members listing groups they are a member of
-	ActiveDirectoryConfig *ActiveDirectoryConfig `json:"activeDirectory,omitempty" description:"schema-specific information for an Active Directory-like schema"`
+	ActiveDirectoryConfig *ActiveDirectoryConfig `json:"activeDirectory" description:"schema-specific information for an Active Directory-like schema"`
 
 	// AugmentedActiveDirectoryConfig holds the configuration for extracting data from an LDAP server
 	// set up in a fashion similar to that used in Active Directory as described above, with one addition:
 	// first-class group entries exist and are used to hold metadata but not group membership
-	AugmentedActiveDirectoryConfig *AugmentedActiveDirectoryConfig `json:"augmentedAD,omitempty" description:"schema-specific information for an Active Directory-like schema with group metadata entries"`
+	AugmentedActiveDirectoryConfig *AugmentedActiveDirectoryConfig `json:"augmentedAD" description:"schema-specific information for an Active Directory-like schema with group metadata entries"`
```

Comment:
> why not omitempty on these three? if we ever write this out, I'd want to omit the unused two

We had previously discussed always generating all options to make user discoverability easier.  It helps people who don't read docs, but if you feel strongly